prowler/providers/aws/services/cognito/cognito_service.py

```diff
@@ -0,0 +1,135 @@
+                    if not self.audit_resources or (
+                        is_resource_filtered(arn, self.audit_resources)
+                    ):
+                        try:
+                            self.user_pools.append(
+                                UserPool(
+                                    id=user_pool["Id"],
+                                    arn=arn,
+                                    name=user_pool["Name"],
+                                    region=regional_client.region,
+                                    last_modified=user_pool["LastModifiedDate"],
+                                    creation_date=user_pool["CreationDate"],
+                                    status=user_pool.get("Status", "Disabled"),
+                                )
+                            )
+                        except Exception as error:
+                            logger.error(
+                                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+                            )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def __describe_user_pools__(self, regional_client):
+        logger.info("Cognito - Describing User Pools...")
+        try:
+            for user_pool in self.user_pools:
+                try:
+                    user_pool_details = regional_client.describe_user_pool(
+                        UserPoolId=user_pool.id
+                    )
+                    user_pool.password_policy = PasswordPolicy(
+                        **user_pool_details.get("Policies", {}).get(
+                            "PasswordPolicy", {}
+                        )
+                    )
+                    user_pool.deletion_protection = user_pool_details.get(
+                        "DeletionProtection", "INACTIVE"
+                    )
+                    user_pool.advanced_security_mode = user_pool_details.get(
+                        "UserPoolAddOns", {}
+                    ).get("AdvancedSecurityMode", "OFF")
+                    user_pool.tags = [user_pool_details.get("UserPoolTags", "")]
+                except Exception as error:
+                    logger.error(
+                        f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+                    )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def __get_user_pool_mfa_config__(self, regional_client):
+        logger.info("Cognito - Getting User Pool MFA Configuration...")
+        try:
+            for user_pool in self.user_pools:
```

Comment:
You need to check if the `user_pool` lives in the same region as the `regional_client` because I'm getting this error `2023-11-28 12:00:05,652 [File: cognito_service.py:101]  [Module: cognito_service]        ERROR: us-east-2 -- ResourceNotFoundException[87]: An error occurred (ResourceNotFoundException) when calling the GetUserPoolMfaConfig operation: User pool eu-west-1_RR2XCbkkm does not exist.`